apps/agreement/contracts/Agreement.sol

```diff
@@ -712,6 +744,46 @@ contract Agreement is IAgreement, AragonApp {
 
     // Internal fns
 
+    /**
+    * @dev Pay transactions fees required for new actions
+    * @param _settingId Identification number of the setting being queried
+    * @param _disputable Address of the Disputable app, used to determine fees
+    * @param _staking Staking pool for the ERC20 token to transfer fees from
+    * @param _submitter Address of the user that has submitted the action
+    * @param _actionId Identification number of the action to be paid for
+    */
+    function _payAppFees(uint256 _settingId, IDisputable _disputable, Staking _staking, address _submitter, uint256 _actionId) internal {
+        // Get fees
+        Setting storage setting = _getSetting(_settingId);
+        IAragonAppFeesCashier aragonAppFeesCashier = setting.aragonAppFeesCashier;
+        if (address(aragonAppFeesCashier) == address(0)) {
+            return;
+        }
+
+        bytes32 appId = _disputable.appId();
+        (ERC20 token, uint256 amount) = aragonAppFeesCashier.getAppFee(appId);
+
+        if (amount == 0) {
+            return;
+        }
+
+        // Get staking pool
+        Staking staking;
+        if (token == _staking.token()) {
+            staking = _staking;
+        } else {
+            staking = stakingFactory.getOrCreateInstance(token);
+        }
+
+        // Get amount from staking
+        _lockBalance(staking, _submitter, amount);
+        _slashBalance(staking, _submitter, address(this), amount);
```

Comment:
Does staking have a way to approve only a certain amount of funds to a lock manager?

Reply:
Yes, there’s the allowance concept for that, so owner allows a maximum amount to be locked by the lock manager.